src/components/Disturber.tsx

```diff
@@ -0,0 +1,186 @@
+import { DrawerNavigationProp } from '@react-navigation/drawer';
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, TouchableOpacity, View } from 'react-native';
+
+import { Icon, colors, normalize } from '../config';
+import { addToStore, findClosestItem, isActive, readFromStore } from '../helpers';
+import { useHomeRefresh, useStaticContent } from '../hooks';
+
+import { Button } from './Button';
+import { Image } from './Image';
+import { BoldText, RegularText } from './Text';
+import { Wrapper, WrapperHorizontal } from './Wrapper';
+
+type Props = {
+  navigation: DrawerNavigationProp<any>;
+  publicJsonFile: string;
+};
+
+interface DateRange {
+  dateEnd: string;
+  dateStart: string;
+}
+
+interface DataItem {
+  backgroundColor?: string;
+  button: {
+    navigationTo: string;
+    params: { title?: string; webUrl: string };
+    title?: string;
+  };
+  headline?: string;
+  dates: DateRange[];
+  description: string;
+  id: number;
+  picture?: { aspectRatio?: { HEIGHT: number; WIDTH: number }; uri: string };
+  title: string;
+}
+
+// eslint-disable-next-line complexity
+export const Disturber = ({ navigation, publicJsonFile }: Props) => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const { data, refetch } = useStaticContent<DataItem[]>({
+    refreshTimeKey: `publicJsonFile-${publicJsonFile}`,
+    name: publicJsonFile,
+    type: 'json'
+  });
+
+  useHomeRefresh(refetch);
+
+  // find the closest item to the current date/time
+  const closestItem: DataItem | null = findClosestItem(
+    data?.filter((item) => item && isActive(item)) || []
+  );
+
+  const setDisturberComplete = () => {
+    setIsVisible((prev) => !prev);
+    addToStore(publicJsonFile, closestItem?.id.toString());
+  };
```

Comment:
line 53: fallback should be `{}` instead of `[]`, or am i wrong? DataItem is not an array.
on the other hand the type is `DataItem | null`, so there is no fallback needed, as it is ok to be null, right?

line 58: if `closestItem` would be an `[]` you wouldn't be able to call `?.id` on it. anyways, if there is no `closestItem` we don't want to add anything to store, right? so we should call `addToStore` just in case we have an item to store?

Reply:
I think you are wrong. Because here I need to give the format that `findClosesItem` expects. `findClosestItem` is expecting an array. I think there should be an empty array. `closestItem` does not have a fallback, we give a value for the `findClosestItem` helper. This value should be an array.

The `setDisturberComplete` function only works when we press the button to close the disturber. If there is no `closestItem`, the disturber does not appear. For this reason, if `closestItem.id` does not exist, there is no disturber. So I think there is no need to add an extra control here.

Reply:
1. oh yes i see about the array. thanks.

2. ok, let's remove the ?

Reply:
Yes, we can remove the ? 👍

Reply:
But in this case, we get an error because closestItem may return null value.
<img width="504" alt="image" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/3416d10b-e1d9-4d5c-bd7d-af6106097af0">

Reply:
could this be fixed with `closestItem && addToStore`?

Reply:
Yeah!! ✌️
https://github.com/smart-village-solutions/smart-village-app-app/pull/909/commits/4dd1f20a139e88749b4a6e145ef93f008948fcab